rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1548,62 +1365,20 @@ def _calculate_label_loss(
             text_embed, label_embed, label_ids, all_labels_embed, all_label_ids
         )
 
-    def _calculate_entity_loss(
-        self,
-        inputs: tf.Tensor,
-        tag_ids: tf.Tensor,
-        mask: tf.Tensor,
-        sequence_lengths: tf.Tensor,
-        tag_name: Text,
-        entity_tags: Optional[tf.Tensor] = None,
-    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]:
-
-        tag_ids = tf.cast(tag_ids[:, :, 0], tf.int32)
-
-        if entity_tags is not None:
-            _tags = self._tf_layers[f"embed.{tag_name}.tags"](entity_tags)
-            inputs = tf.concat([inputs, _tags], axis=-1)
-
-        logits = self._tf_layers[f"embed.{tag_name}.logits"](inputs)
-
-        # should call first to build weights
-        pred_ids, _ = self._tf_layers[f"crf.{tag_name}"](logits, sequence_lengths)
-        loss = self._tf_layers[f"crf.{tag_name}"].loss(
-            logits, tag_ids, sequence_lengths
-        )
-        f1 = self._tf_layers[f"crf.{tag_name}"].f1_score(tag_ids, pred_ids, mask)
-
-        return loss, f1, logits
-
-    @staticmethod
-    def _get_sequence_lengths(
-        tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]],
-        key: Text,
-        sub_key: Text,
-        batch_dim: int = 1,
-    ) -> tf.Tensor:
-        # sentence features have a sequence lengths of 1
-        # if sequence features are present we add the sequence lengths of those
-
-        sequence_lengths = tf.ones([batch_dim], dtype=tf.int32)
-        if key in tf_batch_data and sub_key in tf_batch_data[key]:
-            sequence_lengths += tf.cast(tf_batch_data[key][sub_key][0], dtype=tf.int32)
-
-        return sequence_lengths
-
-    @staticmethod
-    def _get_batch_dim(tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]]) -> int:
-        if TEXT in tf_batch_data and SEQUENCE in tf_batch_data[TEXT]:
-            return tf.shape(tf_batch_data[TEXT][SEQUENCE][0])[0]
-
-        return tf.shape(tf_batch_data[TEXT][SENTENCE][0])[0]
-
     def batch_loss(
         self, batch_in: Union[Tuple[tf.Tensor], Tuple[np.ndarray]]
     ) -> tf.Tensor:
+        """Calculates the loss for the given batch.
+
+        Args:
+            batch_in: The batch.
+
+        Returns:
+            The loss of the given batch.
+        """
         tf_batch_data = self.batch_to_model_data_format(batch_in, self.data_signature)
 
-        batch_dim = self._get_batch_dim(tf_batch_data)
+        batch_dim = self._get_batch_dim(tf_batch_data[TEXT])
```

Comment:
This needs to be changed inside `DIET2DIET` as well.

Reply:
@tabergma could you please take this over